src/sentry/audit_log/manager.py

```diff
@@ -82,21 +80,20 @@ class AuditLogEventManager:
     def __init__(self) -> None:
         self._event_registry = {}
         self._event_id_lookup = {}
+        self._api_name_lookup = {}
 
     def add(self, audit_log_event: AuditLogEvent):
         if (
             audit_log_event.name in self._event_registry
             or audit_log_event.event_id in self._event_id_lookup
         ):
-            capture_exception(
-                DuplicateAuditLogEvent(
-                    f"Duplicate audit log: {audit_log_event.name} with ID {audit_log_event.event_id}"
-                )
+            raise DuplicateAuditLogEvent(
+                f"Duplicate audit log: {audit_log_event.name} with ID {audit_log_event.event_id}"
             )
-            return
 
         self._event_registry[audit_log_event.name] = audit_log_event
         self._event_id_lookup[audit_log_event.event_id] = audit_log_event
+        self._api_name_lookup[audit_log_event.api_name] = audit_log_event
```

Comment:
this will only cache a single audit log per api name.
Would there ever be a case where we'd have multiple audit log events for the same api?

Reply:
@nhsiehgit there should be only a unique 1:1 (bijection) association between `audit_log_event.api_name` and `audit_log_event`.

Reply:
For context, this manager contains all the possible audit logs that can be recorded and the formatting logic for each of them.